src/Test/Parse.idr

```diff
@@ -48,6 +48,10 @@ testSuite = spec $ do
 
     "(+ 1 1 0 (* 1 0))" `itShouldParseAs` "Just (0 ** (1 + 1 + 0 + (1 * 0)))"
 
+    "(mu nat (Z 1) (S (var 0))) (name mnat (+ 1 nat)) (mu listmnat (nil 1) (cons (* mnat (var 0))))"
+      `itShouldParseAs`
+        "Just (0 ** listmnat = mu [nil: 1, cons: ((1 + nat = mu [Z: 1, S: {0}]) * {0})])"
```

Comment:
Just a prettyprinting nitpick, but I think it would aid readability to put `nat = mu [Z: 1, S: {0}]` in brackets. Perhaps `:=` would be more readable than `=` too, e.g.
```
"Just (0 ** listmnat := mu [nil: 1, cons: ((1 + (nat := mu [Z: 1, S: {0}])) * {0})])"
```
? I hope that's an easy change, otherwise it's probably not worth the effort.

Reply:
Not sure if you are going to do this @clayrat but let me know if I need to merge anything.

Reply:
Yes, gonna push this tomorrow